proj/funcDictionary.py

```diff
@@ -0,0 +1,335 @@
+# info: some general functions:
+
+import numpy as np
+import networkx as nx
+
+#import pyspark
+from imnet import process_strings as p  # used?
+
+#import subprocess
+#import re
+#from IPython.display import HTML
+
+#import sparkhpc
+#from sparkhpc import sparkjob
+#from sparkhpc.sparkjob import LSFSparkJob
+#import findspark
+#from scipy.sparse import csr_matrix
+
+#from plotData import PlotData
+#from plotData import InnerDataLevenshtein
+#from plotData import PlotDataLevenshtein
+
+def make_nets(ls, seq): # We still need seq? What is that?
+    """
+    info: make a list "nets", where each element "net" is 
+        a list again, containing all sequences with a certain 
+        length
+    input:
+        ls:  list of length values for each sequence in seq
+        seq: array of nucleic acid sequences 
+    output: 
+        nets: list of nets, where each element "net" is 
+        a list again, containing all sequences with a certain 
+        length  
+    """
+
+    lsHist = np.histogram(ls, bins = np.arange(0, 120, 3))
+    xArgs = lsHist[1][:]
+
+    # keep in mind, that this is the average:
+
+    # info: make empty list of nets:
+    nets = list()
+    for i in range(len(lsHist[0])+1):
+        nets.append(list())
+
+    # info: insert every sequence in the net for its size.
+    #     Thus all sequences with one size are in one net
+    for seqEl in seq:
+        seqElIndex = int((len(seqEl) - lsHist[1][0])/3)
+        nets[seqElIndex].append(seqEl)
+    return nets
+def loadSequence(step, plotData, i = -1, isExtractNum = True):
+    """
+    info: - load the nucleic acid sequences, generated from SONIA 
+        (and stored in a file 'pre.txt' or 'post.txt')
+          - cut of all acids after a certain length, defined by 
+              the threshold "extractNum"
+          - store the sequences in the list "seq" and the lengths of the 
+        the sequences in "ls"
+    input: step: 0 -> data before the Thymus selection (load from pre.txt)
+                 1 -> data after the Thymus selection (load from post.txt)
+        plotData: PlotData object, that contains the important data and parameters;
+            for more detail see documentation in plotData.py
+        i: int number of letters to be extracted from each strain, e.g. if i = 18,
```

Comment:
from each string instead of strain?

Reply:
Yes, with string and strain, I mean the same thing. A strain is just a string of letters. Of course I should decide for
one name, either to say "strain" all the time or to say "string" all the time.

Reply:
I would suggest string, because "strain" is confusing in a biology setting https://en.wikipedia.org/wiki/Strain_(biology)